federation/pkg/dnsprovider/providers/aws/route53/route53.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// route53 is the implementation of pkg/dnsprovider interface for AWS Route53
+package route53
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"io"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+)
+
+const (
+	ProviderName = "aws-route53"
+)
+
+func init() {
+	dnsprovider.RegisterDnsProvider(ProviderName, func(config io.Reader) (dnsprovider.Interface, error) {
+		return newRoute53(config)
+	})
+}
+
+// newRoute53 creates a new instance of an AWS Route53 DNS Interface.
+func newRoute53(config io.Reader) (*Interface, error) {
+	// Connect to AWS Route53 - TODO: Do more sophisticated auth
+	svc := route53.New(session.New( /* &aws.Config{Region: aws.String("us-west-2")}*/ ))
+	// Possibly override defaults with config below
+	if config != nil {
```

Comment:
That was actually a placeholder for actually supporting configuration. But agreed, I can rip it out until we do.